var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -657,36 +658,35 @@ def __init__(self, spec, **kwargs):
         if '+float32' in spec:
             self.precision_option = 'SP'
 
-        # spec.architecture.platform is like `uname -s`, but lower-case
-        platform = spec.architecture.platform
+        # OpenFOAM chooses which wmake rules to use based on the output of
+        # `uname -s` and `uname -m`.  We can get the same information from
+        # Python's platform library.
 
-        # spec.architecture.target is like `uname -m`
-        target   = spec.architecture.target
+        system = platform.system().lower()
+        machine = platform.machine()
 
```

Comment:
From my understanding, `platform.system()` returns the same as `uname -s`, and `platform.machine()` returns the same as `uname -m`.  OpenFOAM uses `uname -s` and `uname -m` to determine its `WM_ARCH*` settings.  So if we want to use spack-specified targets and such for cross-compiling, then we would need to patch OpenFOAM's config files to support it.